tests/svd_solvers.cpp

```diff
@@ -0,0 +1,42 @@
+
+#include <catch.hpp>
+
+#include "solver/svd/svd.hpp"
+
+#include "io/json.hpp"
+#include <iostream>
+#include <chrono>
+
+using namespace neon;
+
+constexpr auto ZERO_MARGIN = 1.0e-5;
+
+TEST_CASE("svd solver test suite")
+{
+    SECTION("singula values and singular vectors")
+    {
+        Eigen::MatrixXd m = Eigen::MatrixXd::Random(10, 50);
+        bdc_svd svd_decomposition;
+        svd_decomposition.compute(m);
+    }
+    SECTION("svd timing")
+    {
+        Eigen::MatrixXd m = Eigen::MatrixXd::Random(100000, 50);
+
+        auto const start = std::chrono::steady_clock::now();
+
+        bdc_svd svd_decomposition;
+        svd_decomposition.compute(m);
+
+        Eigen::MatrixXd m_reconstructed = svd_decomposition.left()
+                                          * svd_decomposition.values().asDiagonal()
+                                          * svd_decomposition.right();
+        Eigen::MatrixXd approximation_error = m - m_reconstructed;
+        std::cout << "approximation error due to svd:\n" << approximation_error.norm() << "\n";
+
+        auto const end = std::chrono::steady_clock::now();
+        std::chrono::duration<double> const elapsed_seconds = end - start;
+
+        std::cout << std::string(6, ' ') << "SVD test case took " << elapsed_seconds.count() << "s\n";
```

Comment:
The six spaces in the print out aren't required here.  That's just a ghetto way of formatting when the simulation is outputting information to the terminal.

Reply:
Actually - this reminds me that we need a good method of performance testing to avoid any regressions - the `ctest` timings aren't the most accurate.